packages/fxa-auth-server/scripts/refund-unverified-accounts.ts

```diff
@@ -0,0 +1,399 @@
+  const header = Object.keys(data[0]);
+  return [
+    header.join(','), // header row first
+    ...data.map((row: OutputData | ErrorOutput) =>
+      header
+        .map((fieldName: string) =>
+          JSON.stringify(row[fieldName as keyof typeof row], replacer)
+        )
+        .join(',')
+    ),
+  ].join('\r\n');
+}
+
+async function writeCsv(data: OutputData[] | ErrorOutput[], file: string) {
+  if (!data.length) {
+    return;
+  }
+  const csv = convertToCsv(data);
+  try {
+    await fs.writeFile(`${file}_${parseDateForFirestore(Date.now())}.csv`, csv);
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+async function readSkip(file: string) {
+  const data = await fs.readFile(file, { encoding: 'utf-8' });
+  return data.split(/\r?\n/);
+}
+
+/**
+ * Retrieve subscriptions by created date
+ */
+async function retrieveSubscriptionsByCreated(
+  startDate: number,
+  endDate: number
+) {
+  const firestore = Container.get(AuthFirestore);
+  const { subscriptionCollection } = getCollectionNames();
+
+  const subscriptionSnap = await firestore
+    .collectionGroup(subscriptionCollection)
+    .where('created', '>=', startDate)
+    .where('created', '<=', endDate)
+    .get();
+  if (!subscriptionSnap.empty) {
+    console.log(subscriptionSnap.docs[0]);
+    return subscriptionSnap.docs.map((doc) => {
+      return {
+        uid: getUidFromPath(doc),
+        subscription: doc.data() as Stripe.Subscription,
+      };
+    });
+  }
+  throw new Error(
+    `No subscriptions found between ${new Date(
+      startDate * 1000
+    ).toDateString()} and ${new Date(endDate * 1000).toDateString()}`
+  );
+}
+
+/**
+ * Retrieve invoice by invoice id
+ */
+async function retrieveInvoice(firestore: Firestore, invoiceId: string) {
+  const { invoiceCollection } = getCollectionNames();
+
+  const invoiceSnapshot = await firestore
+    .collectionGroup(invoiceCollection)
+    .where('id', '==', invoiceId)
+    .get();
+  if (!invoiceSnapshot.empty) {
+    return invoiceSnapshot.docs[0].data() as Stripe.Invoice;
+  }
+  throw new Error(`No invoice found for ${invoiceId}`);
+}
+
+/**
+ * Refund and Delete Customer, Cancel Subscription
+ */
+export async function cancelSubscriptionsAndDeleteCustomer(
+  uid: string,
+  email: string,
+  subscriptionId: string,
+  invoiceId: string
+): Promise<void> {
+  const stripeHelper = Container.get(StripeHelper);
+  const firestore = Container.get(AuthFirestore);
+
+  const invoice = await retrieveInvoice(firestore, invoiceId);
+  const paymentIntent =
+    typeof invoice.payment_intent === 'string'
+      ? invoice.payment_intent
+      : invoice.payment_intent?.id;
+
+  if (!paymentIntent) {
+    throw new Error(`Could not find payment intent for ${email}`);
+  }
+
+  // Cancel any subscriptions and issue refunds
+  await stripeHelper.refundPayment(paymentIntent, 'fraudulent');
+  await stripeHelper.cancelSubscription(subscriptionId);
```

Comment:
I dont think I did this in my script but I am wondering if we should be cancelling the subscription regardless of whether the call to `refundPayment` returns successfully?

If we miss cancelling a subscription because the refund failed for whatever reason, I am guessing they are going to get charged again at some future date.

Reply:
I might not be understanding 100%, but if the refund fails, the process fails for that customer, and we'll be able to re-run the script, which should pick up the failed customer, and try again.

FYI, funnily enough I did actually copy this from your script, I just removed the PayPal stuff, since it wasn't necessary.

Reply:
What I wrote read kinda funny; I recognize that this came from my script and what I meant was what I am about to suggest was something I failed to do when I was doing something similar. 

My suggestion was that we shouldn't make them reliant on each other so that if the refund failed the subscription wouldn't still be active and they wouldn't get subsequent charges, but re-running as you suggested could be another strategy. For any account that fails multiple times, we could just manually handle that in the stripe dashboard.